src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,663 @@
+use std::{
+    cell::UnsafeCell,
+    fmt::Debug,
+    sync::{atomic::AtomicU64, Arc},
+    u8,
+};
+
+use bytes::BytesMut;
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+};
+use tracing::debug;
+
+use crate::{async_fuse::util::usize_to_u64, read_exact_timeout, write_all_timeout};
+
+use super::{
+    common::ClientTimeoutOptions,
+    error::RpcError,
+    message::{ReqType, RespType},
+    packet::{Decode, Encode, Packet, PacketsKeeper, ReqHeader, RespHeader, REQ_HEADER_SIZE},
+    utils::u64_to_usize,
+};
+
+/// TODO: combine `RpcClientConnectionInner` and `RpcClientConnection`
+struct RpcClientConnectionInner<P>
+where
+    P: Packet + Clone + Send + Sync + 'static,
+{
+    /// The TCP stream for the connection.
+    stream: UnsafeCell<TcpStream>,
+    /// Options for the timeout of the connection
+    timeout_options: ClientTimeoutOptions,
+    /// Stream auto increment sequence number, used to mark the request and response
+    seq: AtomicU64,
+    /// Received keep alive seq number, will mark the valid keep alive response.
+    /// If the keep alive response is not received in right mestamp, the connection will be closed,
+    /// If we receive other response, we will update the received_keepalive_timestamp too, just treat it as a keep alive response.
+    received_keepalive_timestamp: AtomicU64,
+    /// Send packet task
+    packets_keeper: UnsafeCell<PacketsKeeper<P>>,
```

Comment:
Remember to change packets_keeper to Mutex

Reply:
Thanks, in `PacketsKeeper` inner, it has a RwLock to make sure race safety.